src/formatDate.js

```diff
@@ -8,7 +8,32 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const fromSeparator = fromFormat[3];
+  const toSeparator = toFormat[3];
+  const dateParts = date.split(fromSeparator);
+  const dateSet = {};
+
+  for (let i = 0; i < dateParts.length; i++) {
+    dateSet[fromFormat[i]] = dateParts[i];
+  }
+
+  const newFormat = [];
+
+  for (const key of toFormat) {
+    if (dateSet[key]) {
+      newFormat.push(dateSet[key]);
+    } else if (key === 'YY') {
+      newFormat.push(dateSet.YYYY.slice(2));
+    } else if (key === 'YYYY') {
+      if (dateSet.YY < 30) {
+        newFormat.push('20' + dateSet.YY);
+      } else {
+        newFormat.push('19' + dateSet.YY);
+      }
+    }
```

Comment:
The function does not handle cases where the date format specified in `fromFormat` does not match the actual format of the `date` string. Consider adding validation to ensure the `date` string matches the `fromFormat` before processing.